### PR TITLE
PE-10480 PE-10840 Added common authoriser

### PR DESCRIPTION
**Pull request by knowvishnu**

### CODE REVIEW

File: superset/peak/authorizer.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            username = 'admin'
+        elif (auth_response['role'] == 'tenantAdmin'):
+            username = 'peakadmin'
+        else:
+            privileges = json.loads(auth_response['privileges'])
+            if has_dashboard_write_access(privileges):
+                username = 'peakuser'
+
+        user = sm.find_user(username)
+        login_user(user, remember=False,
```
Comment (sukhbir-singh) [inline superset/peak/authorizer.py:63]: add condition for logging user based on boolean parameter. take parameter default as true.

**Review by sukhbir-singh — APPROVED** [review]

**Review by abhijeet-peak — APPROVED** [review]

**Review by shreyashfederer — APPROVED** [review]

File: superset/peak/authorizer.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        user = sm.find_user(username)
+
+        # perform login if boolean argument is True
+        if login:
+            login_user(user, remember=False,
+                        duration=timedelta(
+                            auth_response['exp'] - int(
+                                datetime.now().timestamp())))
+
+        logging.info("Successful Login!")
```
Comment (mukulsaini) [inline superset/peak/authorizer.py:66]: Successful login won't give much useful info. better to add the username also here.

File: superset/peak/decorators.py [hunk 3]
```diff
@@ -59,38 +38,13 @@ def check_access_and_create_session(f):
...
         try:
             form_data = request.args.get('form_data')
-            form_data_str = json.loads(form_data)
-            form_auth_token = form_data_str["token"]
-
-            header_auth_token = request.headers.get('Authorization')
-
-            if header_auth_token is not None or form_auth_token is not None:
-                token = header_auth_token
+            form_str = json.loads(form_data)
```
Comment (mukulsaini) [inline superset/peak/decorators.py:41]: this can break if `form_data` is `None`, and we do `json.loads` on `None` type.

**Review by mukulsaini — CHANGES_REQUESTED** [review]

File: superset/peak/authorizer.py (hunk 2 above)
Comment (sukhbir-singh) [reply superset/peak/authorizer.py]: resolved

File: superset/peak/decorators.py (hunk 3 above)
Comment (sukhbir-singh) [reply superset/peak/decorators.py]: resolved

**Review by sukhbir-singh — APPROVED** [review]

**Review by mukulsaini — APPROVED** [review]

**Pull request merged**